To be done

see #24092

Relevant test output before change:
```
=================================================================================================== FAILURES ===================================================================================================
__________________________________________________________________________________ TestLogLocator.test_tick_values_not_empty ___________________________________________________________________________________

self = <matplotlib.tests.test_ticker.TestLogLocator object at 0x000001B75272A7A0>

    def test_tick_values_not_empty(self):
        mpl.rcParams['_internal.classic_mode'] = False
        ll = mticker.LogLocator(subs=(1, 2, 5))
        test_value = np.array([1.e-01, 2.e-01, 5.e-01, 1.e+00, 2.e+00, 5.e+00,
                               1.e+01, 2.e+01, 5.e+01, 1.e+02, 2.e+02, 5.e+02,
                               1.e+03, 2.e+03, 5.e+03, 1.e+04, 2.e+04, 5.e+04,
                               1.e+05, 2.e+05, 5.e+05, 1.e+06, 2.e+06, 5.e+06,
                               1.e+07, 2.e+07, 5.e+07, 1.e+08, 2.e+08, 5.e+08,
                               1.e+09, 2.e+09, 5.e+09])
>       assert_array_equal(ll.tick_values(1, 1e8), test_value)

lib\matplotlib\tests\test_ticker.py:260:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([], dtype=float64), array([1.e-01, 2.e-01, 5.e-01, 1.e+00, 2.e+00, 5.e+00, 1.e+01, 2.e+...5.e+05, 1.e+06, 2.e+06, 5.e+06,
       1.e+07, 2.e+07, 5.e+07, 1.e+08, 2.e+08, 5.e+08, 1.e+09, 2.e+09,
       5.e+09]))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           (shapes (0,), (33,) mismatch)
E            x: array([], dtype=float64)
E            y: array([1.e-01, 2.e-01, 5.e-01, 1.e+00, 2.e+00, 5.e+00, 1.e+01, 2.e+01,
E                  5.e+01, 1.e+02, 2.e+02, 5.e+02, 1.e+03, 2.e+03, 5.e+03, 1.e+04,
E                  2.e+04, 5.e+04, 1.e+05, 2.e+05, 5.e+05, 1.e+06, 2.e+06, 5.e+06,...

C:\Users\David\AppData\Local\Programs\Python\Python310\lib\contextlib.py:79: AssertionError
=========================================================================================== short test summary info ============================================================================================
FAILED lib/matplotlib/tests/test_ticker.py::TestLogLocator::test_tick_values_not_empty - AssertionError:
===================================================================== 1 failed, 7626 passed, 1640 skipped, 13 xfailed in 347.11s (0:05:47) =====================================================================
```

--- a/lib/matplotlib/ticker.py
+++ b/lib/matplotlib/ticker.py
@@ -2386,7 +2386,8 @@ class LogLocator(Locator):
         # Get decades between major ticks.
         stride = (max(math.ceil(numdec / (numticks - 1)), 1)
                   if mpl.rcParams['_internal.classic_mode'] else
-                  (numdec + 1) // numticks + 1)
+                  next(s for s in itertools.count(1)
+                       if numdec // s + 1 <= numticks))
 
         # if we have decided that the stride is as big or bigger than
         # the range, clip the stride back to the available range - 1

--- a/lib/matplotlib/ticker.py
+++ b/lib/matplotlib/ticker.py
@@ -2383,12 +2383,6 @@ class LogLocator(Locator):
         else:
             subs = self._subs
 
-        def alt_stride():
-            s = 1
-            while numdec // s + 1 > numticks:
-                s += 1
-            return s
-
         # Get decades between major ticks.
         stride = (max(math.ceil(numdec / (numticks - 1)), 1)
                   if mpl.rcParams['_internal.classic_mode'] else

--- a/lib/matplotlib/ticker.py
+++ b/lib/matplotlib/ticker.py
@@ -2383,10 +2383,17 @@ class LogLocator(Locator):
         else:
             subs = self._subs
 
+        def alt_stride():
+            s = 1
+            while numdec // s + 1 > numticks:
+                s += 1
+            return s
+
         # Get decades between major ticks.
         stride = (max(math.ceil(numdec / (numticks - 1)), 1)
                   if mpl.rcParams['_internal.classic_mode'] else
-                  (numdec + 1) // numticks + 1)
+                  alt_stride())
+                  # (numdec + 1) // numticks + 1)
 
         # if we have decided that the stride is as big or bigger than
         # the range, clip the stride back to the available range - 1

